django/contrib/sessions/backends/base.py

```diff
@@ -101,12 +106,16 @@ def _hash(self, value):
         return salted_hmac(key_salt, value).hexdigest()
 
     def encode(self, session_dict):
-        "Return the given session dictionary serialized and encoded as a string."
-        serialized = self.serializer().dumps(session_dict)
-        hash = self._hash(serialized)
-        return base64.b64encode(hash.encode() + b":" + serialized).decode('ascii')
+        """Return the given session dictionary serialized and encoded as a string."""
+        return signing.dumps(session_dict, salt=self.key_salt, serializer=self.serializer)
 
     def decode(self, session_data):
+        try:
+            return signing.loads(session_data, salt=self.key_salt, serializer=self.serializer)
+        except Exception:
+            return self._legacy_decode(session_data)
```

Comment:
We should state clearly what to use when deprecation ends, there is a risk that we'll remove logging of suspicious session:
```python
    def decode(self, session_data):
        try:
            return signing.loads(session_data, salt=self.key_salt, serializer=self.serializer)
        # RemovedInDjango40Warning: when the deprecation ends, replace with:
        #   except signing.BadSignature:
        #       logger = logging.getLogger('django.security.SuspiciousSession')
        #       logger.warning('Session data corrupted')
        #   except Exception:
        #       # ValueError, unpickling exceptions. If any of these happen,
        #       # return an empty dictionary (an empty session).
        #       pass
        #   return {}
        except Exception:
            return self._legacy_decode(session_data)
```

Reply:
OK, but IMO some descriptive comment is necessary because it's not a straightforward change, e.g. in `_legacy_decode` we catch `SuspiciousSession` and when the deprecation end we need to catch `signing.BadSignature` etc.